### PR TITLE
PMR tests for indirect and polymorphic

### LINKED ISSUE #17: Add tests with more complicated, real allocators

### PR DESCRIPTION
**Pull request by Twon**
fixes #17

### CODE REVIEW

File: DRAFT.md [hunk 1]
```diff
@@ -282,6 +286,9 @@ class indirect {
   template <class U>
   friend constexpr auto operator<=>(const U& lhs, const indirect<T, A>& rhs);
 };
+
+template <class T, class Alloc>
+struct std::uses_allocator<indirect<T>, Alloc> : true_type {};
```
Comment (jbcoe) [inline DRAFT.md:291]: Q: Should this be `indirect<T, Alloc>` ?

File: DRAFT.md [hunk 2]
```diff
@@ -607,6 +665,9 @@ class polymorphic {
 
   friend constexpr void swap(polymorphic& lhs, polymorphic& rhs) noexcept;
 };
+
+template <class T, class Alloc>
+struct std::uses_allocator<polymorphic<T>, Alloc> : true_type {};
```
Comment (jbcoe) [inline DRAFT.md:670]: Q: Should this be `polymorphic<T, Alloc>` ?

File: DRAFT.md (hunk 1 above)
Comment (Twon) [reply DRAFT.md]: Good catch, now corrected

File: DRAFT.md (hunk 2 above)
Comment (Twon) [reply DRAFT.md]: Also corrected

File: polymorphic_test.cc [hunk 3]
```diff
@@ -405,4 +409,20 @@ TEST(PolymorphicTest, InteractionWithSizedAllocators) {
   EXPECT_EQ(sizeof(xyz::polymorphic<int, TrackingAllocator<int>>),
             (sizeof(int*) + sizeof(TrackingAllocator<int>)));
 }
+
+#if (__cpp_lib_memory_resource >= 201603L)
+TEST(PolymorphicTest, InterationWithPMRAllocators) {
```
Comment (jbcoe) [inline polymorphic_test.cc:414]: typo: InteractionWithPMRAllocators

File: indirect_test.cc [hunk 4]
```diff
@@ -432,4 +436,18 @@ TEST(IndirectTest, InteractionWithSizedAllocators) {
             (sizeof(int*) + sizeof(TrackingAllocator<int>)));
 }
 
+#if (__cpp_lib_memory_resource >= 202207L)
+TEST(IndirectTest, InterationWithPMRAllocators)
```
Comment (jbcoe) [inline indirect_test.cc:440]: typo: InteractionWithPMRAllocators

**Review by jbcoe — CHANGES_REQUESTED** [review]

File: indirect_test.cc [hunk 5]
```diff
@@ -432,4 +436,18 @@ TEST(IndirectTest, InteractionWithSizedAllocators) {
...
+  std::pmr::monotonic_buffer_resource mbr{buffer.data(), buffer.size()};
+  std::pmr::polymorphic_allocator<int> pa{&mbr};
+  using IndirectInt = xyz::indirect<int, std::pmr::polymorphic_allocator<int>>;
+  IndirectInt a(std::allocator_arg, pa, std::in_place, 42);
+  std::pmr::vector<IndirectInt> values{pa};
+  values.push_back(a);
+  values.push_back(std::move(a));
+  EXPECT_EQ(values[0]->value(), 42);
+}
+#endif
```
Comment (jbcoe) [inline indirect_test.cc:452]: Nit: #endif // (__cpp_lib_memory_resource >= 202207L)

File: polymorphic_test.cc [hunk 6]
```diff
@@ -405,4 +409,20 @@ TEST(PolymorphicTest, InteractionWithSizedAllocators) {
...
+  std::pmr::polymorphic_allocator<Base> pa{&mbr};
+  using PolymorphicBase = xyz::polymorphic<Base, std::pmr::polymorphic_allocator<Base>>;
+  PolymorphicBase a(
+      std::allocator_arg, pa, std::in_place_type<Derived>, 42);
+  std::pmr::vector<PolymorphicBase> values{pa};
+  values.push_back(a);
+  values.push_back(std::move(a));
+  EXPECT_EQ(values[0]->value(), 42);
+}
+#endif
```
Comment (jbcoe) [inline polymorphic_test.cc:426]: Nit: #endif // (__cpp_lib_memory_resource >= 202207L)

**Review by jbcoe — APPROVED** [review]

**Pull request merged**